block-producer/src/block.rs

```diff
@@ -19,7 +19,7 @@ use crate::store::BlockInputsError;
 pub struct Block {
     pub header: BlockHeader,
     pub updated_accounts: Vec<AccountUpdateDetails>,
-    pub created_notes: Vec<(usize, usize, NoteEnvelope)>,
+    pub created_notes: Vec<NoteCreated>,
```

Comment:
Related to the previous comment: this could have probably have been something like `Vec<(usize, usize, NoteEnvelope)>`. In general, I think we should try to avoid using protobuf types inside domain types.

A thought for the future: we should probably move the `Block` struct into `miden-base` at some point.